pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,201 @@
+package logger
+
+import (
+	"container/list"
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+
+var UnixErrorLogPath = "/opt/aws/aws-otel-collector/logs/aws-otel-collector-error.log"
+
+var WindowsErrorLogPath = "C:\\ProgramData\\Amazon\\AWSOTelCollector\\Logs\\aws-otel-collector-error.log"
+
+var errorLogfile = getErrorLogFilePath()
+
+var errorLogMaxAge = 10
+
+var queue = list.New()
+var errorLogChannel = make(chan zapcore.Entry, 10)
+var ticker = time.NewTicker(10 * time.Minute)
+var quit = make(chan struct{})
+var kilobyte = 1024
+var errorFile *os.File
+
+// WriteError func could write the new error log message into file
+func WriteError(newErrorLog zapcore.Entry, errorLogfile string) (n int, err error) {
+	var byteNum int
+	if errorFile == nil {
+		if err = openExistingOrNewErrorFile(); err != nil {
+			return 0, err
+		}
+	}
+	err = errorFile.Truncate(0)
+	if err != nil {
+		return byteNum, err
+	}
+	// iterate errors in queue, if the old error's message same as the new error's message, skip it,
+	// otherwise write it into the file and queue
+	var backupQueue = list.New()
+	for queue.Len() > 0 {
+		e := queue.Front().Value.(zapcore.Entry)
```

Comment:
I don't think you need to create two queues and dequeue from one into the other. That seems very inefficient. 

This is not how the documentation for list suggests you iterate over it: https://golang.org/pkg/container/list/

Also, more importantly, in go we generally use slices instead of lists. The core list type is only really useful if you need a collection that can hold multiple types of data I think. See: https://stackoverflow.com/questions/21326109/why-are-lists-used-infrequently-in-go

I'd prefer to see this code re-written to use a slice.

Reply:
sure, i will take a look